workflows/callingcards.nf

```diff
@@ -0,0 +1,258 @@
+include { PREPARE_READS as MAMMALS_PREPARE_READS           } from "../subworkflows/local/mammals/prepare_reads.nf"
+include { ALIGN                                            } from "../subworkflows/local/align.nf"
+include { PROCESS_ALIGNMENTS as YEAST_PROCESS_ALIGNMENTS   } from "../subworkflows/local/yeast/process_alignments.nf"
+include { PROCESS_ALIGNMENTS as MAMMALS_PROCESS_ALIGNMENTS } from "../subworkflows/local/mammals/process_alignments.nf"
+include { MULTIQC                                          } from "../modules/nf-core/multiqc"
+include { paramsSummaryMap                                 } from 'plugin/nf-validation'
+include { paramsSummaryMultiqc                             } from '../subworkflows/nf-core/utils_nfcore_pipeline'
+include { softwareVersionsToYAML                           } from '../subworkflows/nf-core/utils_nfcore_pipeline'
+include { methodsDescriptionText                           } from '../subworkflows/local/utils_nfcore_callingcards_pipeline'
+
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    RUN MAIN WORKFLOW
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+workflow CALLINGCARDS {
+
+    take:
+    ch_reads
+    ch_barcode_details
+    ch_fasta
+    ch_gtf
+    ch_regions_mask
+    additional_fasta
+    rseqc_modules
+
+    main:
+
+    ch_versions          = Channel.empty()
+    ch_multiqc_files     = Channel.empty()
+    ch_fasta_index       = Channel.empty()
+    ch_bam_index         = Channel.empty()
+    ch_prepared_reads    = Channel.empty()
+    ch_samtools_stats    = Channel.empty()
+    ch_samtools_flatstat = Channel.empty()
+    ch_samtools_idxstats = Channel.empty()
+
+
+    // SUBWORKFLOW_1: Index the genome in various ways
+    //
+    // input: fasta file (genome), a regions mask (bed format), additional
+    //        sequences (fasta format) to append to the genome after masking,
+    //        and a gtf file
+    // output: fasta (masked fasta), fai (fasta index),
+    //         genome_bed (gtf in bed format), bwamem2_index (bwa mem2 index),
+    //         bwa_index (bwa aln index), bowtie_index (bowtie index),
+    //         bowtie2_index (bowtie2 index), versions
+    //         NOTE: the aligner index channels will be empty except for the
+    //         aligner specified in params.aligner
+    PREPARE_GENOME(
+        ch_fasta,
+        ch_regions_mask,
+        additional_fasta,
+        ch_gtf
+    )
+    ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)
+
+    // SUBWORKFLOW_2: Prepare the reads (trimming, removing callingcards specific barcodes)
+    if(params.datatype == 'yeast'){
+        YEAST_PREPARE_READS (
+            ch_reads,
+            ch_barcode_details
+        )
+
+        ch_versions = ch_versions.mix(YEAST_PREPARE_READS.out.versions)
+
+        ch_multiqc_files = ch_multiqc_files.mix(YEAST_PREPARE_READS.out.raw_fastqc_zip.collect{it[1]}.ifEmpty([]))
```

Comment:
Can you confirm your multiQC report looks as expected, and that all sections are in there? I've had issues in the past with mixing MultiQC files alongside the invocation of the modules, and files got 'lost'. I end up having to put the mixing just before `MULTIQC()` line basically. It may depend on the way the pipeline is written, but it might be good to check.

I also don't know if you've customised your MultiQC config yet either, as it's not in this PR (see the main comment)

Reply:
The multiqc report is displaying as expected, but that is good to know and I will keep an eye on it. I like the `mix` where it is happening b/c the process out output are near one another. I'm going to leave it...against what I recognize is your better advice.

The multiqc modification amounts to adding this:

```
report_section_order:
  "nf-core-callingcards-methods-description":
    order: -1000
  software_versions:
    order: -1001
  "nf-core-callingcards-summary":
    order: -1002
```
So, very minimal. I'm happy with it for now. But, adding assay specific QC metrics to the multiQC report is on my my todo list for a future release